crates/bevy_log/src/lib.rs

```diff
@@ -130,51 +198,98 @@ impl Plugin for LogPlugin {
 
-            // bevy_render::renderer logs a `tracy.frame_mark` event every frame
-            // at Level::INFO. Formatted logs should omit it.
-            #[cfg(feature = "tracing-tracy")]
-            let fmt_layer =
-                fmt_layer.with_filter(tracing_subscriber::filter::FilterFn::new(|meta| {
-                    meta.fields().field("tracy.frame_mark").is_none()
-                }));
+                // bevy_render::renderer logs a `tracy.frame_mark` event every frame
+                // at Level::INFO. Formatted logs should omit it.
+                #[cfg(feature = "tracing-tracy")]
+                let fmt_layer =
+                    fmt_layer.with_filter(tracing_subscriber::filter::FilterFn::new(|meta| {
+                        meta.fields().field("tracy.frame_mark").is_none()
+                    }));
 
-            let subscriber = subscriber.with(fmt_layer);
+                let subscriber = subscriber.with(fmt_layer);
+
+                #[cfg(feature = "tracing-chrome")]
+                let subscriber = subscriber.with(chrome_layer);
+                #[cfg(feature = "tracing-tracy")]
+                let subscriber = subscriber.with(tracy_layer);
+                subscriber
+            };
 
-            #[cfg(feature = "tracing-chrome")]
-            let subscriber = subscriber.with(chrome_layer);
-            #[cfg(feature = "tracing-tracy")]
-            let subscriber = subscriber.with(tracy_layer);
+            let file_appender_layer = if let Some(settings) = &self.file_appender_settings {
+                if settings.use_panic_hook {
+                    let old_handler = std::panic::take_hook();
+                    std::panic::set_hook(Box::new(move |panic_info| {
+                        if let Some(s) = panic_info.payload().downcast_ref::<&str>() {
+                            error!("panic occurred: {s:?}");
+                        } else {
+                            error!("panic occurred");
+                        }
+                        old_handler(panic_info);
+                    }));
+                }
+
+                if settings.rolling == Rolling::Never && settings.prefix.is_empty() {
+                    panic!("Using the Rolling::Never variant with no prefix will result in an empty filename which is invalid");
+                }
+                let file_appender = tracing_appender::rolling::RollingFileAppender::new(
+                    settings.rolling.into(),
+                    &settings.path,
+                    &settings.prefix,
+                );
+
+                let (non_blocking, worker_guard) = tracing_appender::non_blocking(file_appender);
+                // WARN We need to keep this somewhere so it doesn't get dropped.
+                // If it gets dropped then it will silently stop writing to the file
+                app.insert_resource(FileAppenderWorkerGuard(worker_guard));
```

Comment:
This isn't fully robust as it will mean that logs still get lost after teardown of the current app. This would apply to logs generated later in the teardown, or perhaps in a multi-`App` process if the `App` with the logging configuration were to be destroyed but others were to continue.

Given that only one logger can ever be configured globally, perhaps it's okay to simply `mem::forget` the guard? This applies equally to the existing guards elsewhere in the code.